tools/make.py

```diff
@@ -38,157 +38,153 @@
 from tools.paths import UBLOX_LIBRARY
 from tools.tests import TESTS, Test, TEST_MAP
 from tools.tests import TEST_MBED_LIB
+from tools.tests import test_known, test_name_known
 from tools.targets import TARGET_MAP
 from tools.options import get_default_options_parser
 from tools.build_api import build_project
 from tools.build_api import mcu_toolchain_matrix
-try:
-    import tools.private_settings as ps
-except:
-    ps = object()
+from utils import argparse_filestring_type
+from argparse import ArgumentTypeError
 
 if __name__ == '__main__':
     # Parse Options
     parser = get_default_options_parser()
-    parser.add_option("-p",
-                      type="int",
+    group = parser.add_mutually_exclusive_group(required=True)
```

Comment:
To add this comment, here are the valid combinations of arguments for make.py (the top level bullets are exclusive):

- `-L <--list-tests>` - This should not require any other arguments to be set (ex. `-m`, `-t`, etc)
- `-S <--supported-toolchains>` - This should not require any other arguments to be set (ex. `-m`, `-t`, etc)
  - **NOTE:** This command also has historically accepted the `-f <--filter>` option as well, so this capability should not be lost if at all possible.
- One of the following commands (exclusively). All of the following should require all the build-related arguments (ex. `-m`, `-t`, etc):
  - Exactly one instance of `-n`
  - Exactly one instance of `-p`
  - At least one instance of `--source` (potentially more than one)